src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,897 @@
+                With FileInfoFromSystemIO
+                    Assert.True(info.Exists)
+                    Assert.True(info.Attributes = .Attributes)
+                    Assert.True(info.CreationTime > Date.MinValue, "Starttime = 0")
+                    Assert.True(info.DirectoryName = .DirectoryName)
+                    Assert.True(info.Extension = .Extension)
+                    Assert.True(info.FullName = .FullName)
+                    Assert.True(info.IsReadOnly = .IsReadOnly)
+                    Assert.True(info.LastAccessTime = .LastAccessTime)
+                    Assert.True(info.LastWriteTime = .LastWriteTime)
+                    Assert.True(info.Length = .Length)
+                    Assert.True(info.Name = .Name)
+                End With
+
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetFiles_Directory()
+            Using TestBase As New FileIOTests
+                Dim FileList As ReadOnlyCollection(Of String) = FileSystem.GetFiles(TestBase.TestDirectory)
+                Assert.True(FileList.Count = 0)
+                For i As Integer = 0 To 5
+                    CreateTestFile(TestBase, SourceData, TestFileName:=$"NewFile{i}")
+                Next
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory)
+                Assert.True(FileList.Count = 6)
+                For i As Integer = 0 To 5
+                    Assert.True(FileList.Contains(IO.Path.Combine(TestBase.TestDirectory, $"NewFile{i}")))
+                Next
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, "GetFiles_DirectoryNewSubDirectory"))
+                CreateTestFile(TestBase, SourceData, PathFromBase:="GetFiles_DirectoryNewSubDirectory", TestFileName:="NewFile")
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory)
+                Assert.True(FileList.Count = 6)
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetFiles_Directory_SearchOption()
+            Using TestBase As New FileIOTests
+                Dim FileList As ReadOnlyCollection(Of String) = FileSystem.GetFiles(TestBase.TestDirectory)
+                Assert.True(FileList.Count = 0)
+                For i As Integer = 0 To 5
+                    CreateTestFile(TestBase, SourceData, TestFileName:=$"NewFile{i}")
+                Next
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+                Assert.True(FileList.Count = 6)
+                For i As Integer = 0 To 5
+                    Assert.True(FileList.Contains(IO.Path.Combine(TestBase.TestDirectory, $"NewFile{i}")))
+                Next
+                Dim NewSubDirectoryPath As String = IO.Path.Combine(TestBase.TestDirectory, "GetFiles_Directory_SearchOptionNewSubDirectory")
+                IO.Directory.CreateDirectory(NewSubDirectoryPath)
+                CreateTestFile(TestBase, SourceData, TestFileName:="NewFile")
```

Comment:
Please add the sub directory and file before the earlier call to `FileSystem.GetFiles` so it's clear `SearchTopLevelOnly` ignores sub directories.

Reply:
Fixed